tests/commands/global_test.py

```diff
@@ -116,3 +117,64 @@ def test_apply_sign_by_default_but_no_key(self):
 
         self.assertFalse(envelope.sign)
         self.assertIs(envelope.sign_key, None)
+
+
+class TestExternalCommand(unittest.TestCase):
+
+    class Success(Exception):
+        pass
+
+    def on_success(self):
+        raise self.Success
+
+    def test_no_spawn_no_stdin_success(self):
+        cmd = g_commands.ExternalCommand(
+            u'python -c "import sys; sys.exit(0)"',
+            refocus=False, on_success=self.on_success)
+        with self.assertRaises(self.Success):
+            cmd.apply(mock.Mock())
+
+    def test_no_spawn_stdin_success(self):
+        cmd = g_commands.ExternalCommand(
+            u'python -c "import sys; sys.exit(int(sys.stdin.read().strip()))"',
+            stdin=u'0', refocus=False, on_success=self.on_success)
+        with self.assertRaises(self.Success):
+            cmd.apply(mock.Mock())
+
+    def test_no_spawn_failure(self):
+        ui = mock.Mock()
+        cmd = g_commands.ExternalCommand(
+            u'python -c "import sys; sys.exit(1)"',
+            refocus=False, on_success=self.on_success)
+        cmd.apply(ui)
+        ui.notify.assert_called_once_with('', priority='error')
+
+    @mock.patch('alot.commands.globals.settings.get', mock.Mock(return_value=''))
```

Comment:
Is there a reason why you patch `settings.get` in a decorator and `os.environ` in a with block?